src/features/extract_from_files.py

```diff
@@ -87,99 +119,91 @@ def _extract_docx(self, filename: str) -> dict:
             elif RETURN_IMAGES_IN_METADATA and file.filename.find("media") >= 0:
                 images.update(self._extract_image_content(document, file))
 
-        return {"extracted_content": extracted_content, "images": images}
+        return extracted_content, images
 
-    def _get_pdf_content(self, filename: str, pdf_file: PyPDF2.PdfFileReader) -> str:
-        extracted_content = f"{pdf_file.getDocumentInfo()}"
-        data = pdf_file.getXmpMetadata()
+    def _get_pdf_content(self, file: typing.BinaryIO, pdf_file: PyPDF2.PdfFileReader) -> str:
+        extracted_content = f"{pdf_file.metadata}"
+        data = pdf_file.xmp_metadata
         for parameter in self.xmp_metadata:
             if parameter:
                 try:
                     extracted_content += f"{parameter}, {getattr(data, parameter)}"
                 except (AttributeError, TypeError) as err:
-                    self._logger.exception(
-                        f"Parameter in pdf content failed to be retrieved: {parameter} with {err.args}"
-                    )
-        extracted_content += extract_text(filename)
+                    logger.exception(f"Parameter in pdf content failed to be retrieved: {parameter} with {err.args}")
+        extracted_content += extract_text(file)
         return extracted_content
 
     @staticmethod
-    def _get_pdf_images(pdf_file: PyPDF2.PdfFileReader) -> list:
+    def _get_pdf_images(reader: PyPDF2.PdfFileReader) -> list:
         images = []
         resources_tag = "/Resources"
         xobject_tag = "/XObject"
         subtype_tag = "/Subtype"
-        for page in range(pdf_file.getNumPages()):
-            pdf_page = pdf_file.getPage(page)
-            if resources_tag in pdf_page.keys() and xobject_tag in pdf_page[resources_tag].keys():
-                x_object = pdf_page[resources_tag][xobject_tag].getObject()
+        for page in reader.pages:
```

Comment:
This reads like a great candidate for list comprehensions :D Potentially, that is a faster option here.

Reply:
Mhh, could change it to that:

```python

    @staticmethod
    def _get_pdf_images(reader: PyPDF2.PdfFileReader) -> list:
        resources_tag = "/Resources"
        xobject_tag = "/XObject"
        subtype_tag = "/Subtype"

        def images(x_object) -> Iterable[str]:
            for obj in x_object:
                if x_object[obj][subtype_tag] == "/Image":
                    yield obj

        def x_objects():
            for page in reader.pages:
                if resources_tag in page.keys() and xobject_tag in page[resources_tag].keys():
                    yield page[resources_tag][xobject_tag].get_object()

        return [image for x_object in x_objects() for image in images(x_object)]
        
```
not sure if that's better :p

Reply:
changed to generators :)